# Exploratory Data Analysis on dataset given

## Import packages

In [1]:
import os
import numpy as np
import pandas as pd

In [2]:
print(os.getcwd())

D:\Data Science\kaggle\HomeCreditDefaultRisk\notebook


In [3]:
os.chdir('D:/Data Science/kaggle/HomeCreditDefaultRisk')

## Load the base training set

In [4]:
training_base = pd.read_csv('./data/application_train.csv')
print(f'Training Set: {training_base.shape}')

Training Set: (307511, 122)


In [5]:
training_base.head()

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   
2      100004       0    Revolving loans           M            Y   
3      100006       0         Cash loans           F            N   
4      100007       0         Cash loans           M            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   
2               Y             0           67500.0    135000.0       6750.0   
3               Y             0          135000.0    312682.5      29686.5   
4               Y             0          121500.0    513000.0      21865.5   

              ...              FLAG_DOCUMENT_18 FLAG_DOCUMENT_19  \
0             ...                             0                0   
1             ...                             0                0   
2             ...                             0                0   
3             ...                             0                0   
4             ...                             0                0   

  FLAG_DOCUMENT_20 FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR  \
0                0                0                        0.0   
1                0                0                        0.0   
2                0                0                        0.0   
3                0                0                        NaN   
4                0                0                        0.0   

  AMT_REQ_CREDIT_BUREAU_DAY  AMT_REQ_CREDIT_BUREAU_WEEK  \
0                       0.0                         0.0   
1                       0.0                         0.0   
2                       0.0                         0.0   
3                       NaN                         NaN   
4                       0.0                         0.0   

   AMT_REQ_CREDIT_BUREAU_MON  AMT_REQ_CREDIT_BUREAU_QRT  \
0                        0.0                        0.0   
1                        0.0                        0.0   
2                        0.0                        0.0   
3                        NaN                        NaN   
4                        0.0                        0.0   

   AMT_REQ_CREDIT_BUREAU_YEAR  
0                         1.0  
1                         0.0  
2                         0.0  
3                         NaN  
4                         0.0  

[5 rows x 122 columns]

122 columns in base dataframe, let's check the data type and sample values in first row.

In [6]:
for _ in training_base.columns:
    print(f'{_ :50}| {training_base.loc[0, _]}')

SK_ID_CURR                                        | 100002
TARGET                                            | 1
NAME_CONTRACT_TYPE                                | Cash loans
CODE_GENDER                                       | M
FLAG_OWN_CAR                                      | N
FLAG_OWN_REALTY                                   | Y
CNT_CHILDREN                                      | 0
AMT_INCOME_TOTAL                                  | 202500.0
AMT_CREDIT                                        | 406597.5
AMT_ANNUITY                                       | 24700.5
AMT_GOODS_PRICE                                   | 351000.0
NAME_TYPE_SUITE                                   | Unaccompanied
NAME_INCOME_TYPE                                  | Working
NAME_EDUCATION_TYPE                               | Secondary / secondary special
NAME_FAMILY_STATUS                                | Single / not married
NAME_HOUSING_TYPE                                 | House / apartment
REGION_POPULAT

### Some categorical data to check at first glance

In [7]:
interesting_categorical_var = [
    'CODE_GENDER',
    'NAME_EDUCATION_TYPE',
    'OCCUPATION_TYPE',
    'WEEKDAY_APPR_PROCESS_START',
    'NAME_INCOME_TYPE',
    'NAME_FAMILY_STATUS',
]

In [8]:
for column in interesting_categorical_var:
    print(f'Checking {column}...')
    result_df = training_base[[column, 'TARGET']].groupby(column).agg({
        len,
        sum,
    })
    result_df.columns=['Total', 'Defect']
    result_df['Rate'] = result_df.apply(
        lambda _: _['Defect'] / _['Total'] if _['Total'] > 0 else np.nan,
        axis=1
    )
    print(result_df.sort_values(by=['Rate', 'Total'], ascending=[False, False]))
    print('-'*50)

Checking CODE_GENDER...
              Total  Defect      Rate
CODE_GENDER                          
M            105059   10655  0.101419
F            202448   14170  0.069993
XNA               4       0  0.000000
--------------------------------------------------
Checking NAME_EDUCATION_TYPE...
                                Total  Defect      Rate
NAME_EDUCATION_TYPE                                    
Lower secondary                  3816     417  0.109277
Secondary / secondary special  218391   19524  0.089399
Incomplete higher               10277     872  0.084850
Higher education                74863    4009  0.053551
Academic degree                   164       3  0.018293
--------------------------------------------------
Checking OCCUPATION_TYPE...
                       Total  Defect      Rate
OCCUPATION_TYPE                               
Low-skill Laborers      2093     359  0.171524
Drivers                18603    2107  0.113261
Waiters/barmen staff    1348     152  0.1127

Some findings here:

- Males have less application but higher defect rate
- Person with less education have higher defect rate
- Widow, surprisingly have the lower defect rate
- Person applied the loan during weekdays seem to have little bit higher defect rate
- Person with lower skill level at work have lower defect rate